src/Middleware/Diagnostics.Abstractions/src/IStatusCodeReExecuteFeature.cs

```diff
@@ -28,6 +28,11 @@ public interface IStatusCodeReExecuteFeature
     /// </summary>
     string? OriginalQueryString { get; set; }
 
+    /// <summary>
+    /// The <see cref="HttpResponse.StatusCode"/> of the original response.
+    /// </summary>
+    int OriginalStatusCode { get; set; }
```

Comment:
Can this be done in a non-breaking way with DIM's? Not sure how to do that with properties, but the doc claims it's possible.
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/default-interface-methods
```suggestion
    public int OriginalStatusCode { get; set; } = StatusCodes.Status200Ok;
```

Reply:
Would it be better to make it nullable?  200 would be confusing in scenarios where the original status code was something else.

Reply:
I guess you could use -1 or something if you were _really_ opposed to boxing.

Reply:
Also, do people really implement this interface?  How would they use their own implementations?

Reply:
I've updated the API proposal.

Reply:
We decided to make it throw.